src/pybind/mgr/nfs/export.py

```diff
@@ -167,17 +167,49 @@ class AppliedExportResults:
     def __init__(self) -> None:
         self.changes: List[Dict[str, str]] = []
         self.has_error = False
+        self.exceptions: List[Exception] = []
+        self.faulty_export_block_indices: List[int] = []
+        self.status = ""
 
-    def append(self, value: Dict[str, str]) -> None:
+    def append(self, value: Dict[str, Any]) -> None:
         if value.get("state", "") == "error":
+            # If there is an error then there must be an exception in the dict.
+            self.exceptions.append(value.pop("exception"))
+            # Index is for indicating at which export block in the conf/json
+            # file did the export creation/update failed.
+            self.faulty_export_block_indices.append(value.pop("index"))
             self.has_error = True
         self.changes.append(value)
 
     def to_simplified(self) -> List[Dict[str, str]]:
         return self.changes
 
     def mgr_return_value(self) -> int:
-        return -errno.EIO if self.has_error else 0
+        if self.has_error:
+            if len(self.exceptions) == 1:
+                ex = self.exceptions[0]
+                if isinstance(ex, NFSException):
+                    return ex.errno
+                # Some non-nfs exception occurred, this can be anything
+                # therefore return EAGAIN as a generalised errno.
+                return -errno.EAGAIN
+            # There are multiple failures so returning EIO as a generalised
+            # errno.
+            return -errno.EIO
+        return 0
+
+    def mgr_status_value(self) -> str:
+        if self.has_error:
+            if len(self.faulty_export_block_indices) == 1:
+                self.status = f"{str(self.exceptions[0])} for export block" \
+                              f" at index {self.faulty_export_block_indices[0]}"
+            if len(self.faulty_export_block_indices) > 1:
+                indices = ", ".join([str(faulty_block) for faulty_block in
+                                     self.faulty_export_block_indices])
+                self.status = f"{len(self.faulty_export_block_indices)} export" \
+                              f" blocks (at index {indices}) failed to be" \
+                              " created/updated"
+        return self.status
```

Comment:
@vshankar the above code is for the err string(which was missing previously) that will be printed along with errno. If you have any suggestions or changes to request do let me know.

Some examples:
```
Export ID changed, Cannot update export for export block at index 1
```


```
2 export blocks (at index 1, 3) failed to be created/updated
```

Reply:
> @vshankar the above code is for the err string(which was missing previously) that will be printed along with errno. If you have any suggestions or changes to request do let me know.
> 
> Some examples:
> 
> ```
> Export ID changed, Cannot update export for export block at index 1
> ```
> 
> ```
> 2 export blocks (at index 1, 3) failed to be created/updated
> ```

Will "index" be useful to the user in this error string?

Reply:
the indices are intended to help figure out which block(s) in the cfg file (conf/json) failed to be created or updated

Reply:
Anything other "id"entifier that's a _bit_ more self-descriptive?

Reply:
since this is meant to point out to problematic block(s) in the config file, and considering that the blocks might/might not have expord_id's provided, the only identifier that i can think of could be pseudo path

Reply:
or path, but since the path could anything like `/foo/bar/baz/...`, the err reporting string would look ugly

Reply:
Formatting that with json would take care of the ugliness, isn't it? WDYT?

Reply:
this is err status string, we aren't returning json, if json is to be returned it would be something like this  
```
EINVAL:
{
    "paths": foo/bar, bar/baz
}
```

Reply:
Why would there be multiple paths? path+pseudo_path?

Reply:
the conf file can have multiple export blocks

Reply:
Sure, but then we can have multiple entries in json. Wouldn't that suffice?

Reply:
like how

Reply:
(discussion cotd. in slack in the interest of time)